app/controllers/tags_controller.rb

```diff
@@ -18,7 +18,11 @@ class TagsController < ApplicationController
   def index
     skip_authorization
     @tags_index = true
-    @tags = Tag.direct.includes(:sponsorship).order(hotness_score: :desc).limit(100)
+    @tags = if params.key?(:q) && params[:q].present?
+              Tag.includes([:sponsorship]).search_by_name(params[:q]).limit(100)
```

Comment:
Question: should we move the LIMIT into a constant so that we can keep it consistent between both branches without having to make two changes?

Reply:
Another option would be:

```ruby
@tags = Tag.includes([:sponsorship]).limit(100)
if params[:q].present?
  @tags = @tags.search_by_name(params[:q])
else
  @tags = @tags.order(hotness_score: :desc)
end
```

Throughout the code-base we have several places that uses and replaces to an initial scope.

Reply:
I think I like that @jeremyf's suggestion a tad better. in fact it's what I had initially but as `before_action` that initially defined and memoized `@tags` for the index action. I may throw up a commit with this just to collect feedback.